external-crates/move/crates/move-trace-format/src/trace_state_machine.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) The Move Contributors
```

Comment:
A small nit, but I don't think is is a state machine in the traditional sense. I was expecting something more like an automata based on that. Consider renaming this to `trace_state.rs` or similar.

Reply:
Sounds good! Yea this is more of a "state machine" in the sense that it's a "state of the VM tracking machine". I like memory tracer for the name!